### PR TITLE
CoWSwap integration

### PR DESCRIPTION
**Pull request by juliopavila**
### Feature Request

Zodiac Pilot Extension doesn't work with CowSwap and Roles currently, as it needs a wrapper contract so we can adequately scope roles to cowswap orders.

### CODE REVIEW

File: extension/src/transactionTranslations/cowswapSetPreSignature.ts [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+
+export const COWSWAP_SUPPORTED_NETWORK: Record<number, string> = {
+  1: 'mainnet',
+  5: 'goerli',
+  100: 'xdai',
+}
+
+const API_URL = `https://api.cow.fi/`
+
+const abi = [
```
Comment (jfschwarz) [inline extension/src/transactionTranslations/cowswapSetPreSignature.ts:16]: let's strip down the API to only what we need (just this one function)

File: extension/package.json [hunk 2]
```diff
@@ -25,6 +25,7 @@
   },
   "devDependencies": {
     "@gnosis.pm/zodiac": "^2.0.1",
+    "@cowprotocol/cow-sdk": "^2.0.0-alpha.4",
```
Comment (jfschwarz) [inline extension/package.json:28]: we don't seem to use it. let's clean the dependency out

File: extension/src/transactionTranslations/cowswapSetPreSignature.ts [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+  },
+]
+const provider = window.ethereum && new Web3Provider(window.ethereum)
+const cowSwapFunctionSignature = 'setPreSignature(bytes,bool)'
+const cowSwapEncodeFunctionSignature = ethers.utils
+  .id(cowSwapFunctionSignature)
+  .substring(0, 10)
+
+export default {
+  title: 'setPreSignature call',
```
Comment (jfschwarz) [inline extension/src/transactionTranslations/cowswapSetPreSignature.ts:277]:
```suggestion
  title: 'Route through CowswapOrderSigner',
```

File: extension/src/transactionTranslations/cowswapSetPreSignature.ts (hunk 1 above)
Comment (jfschwarz) [reply extension/src/transactionTranslations/cowswapSetPreSignature.ts:13]: is it not possible to minimize this json?

basically to only:

```
const abi = [
  {
     type: 'function',
     name: 'setPreSignature',
     constant: false,
     payable: false,
     inputs: [
       { type: 'bytes', name: 'orderUid' },
       { type: 'bool', name: 'signed' },
     ],
     outputs: [],
   },
]
```

**Review by jfschwarz — APPROVED** [review]
🚀

File: extension/src/transactionTranslations/cowswapSetPreSignature.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            try {
+              const COW_SWAP_URL = `https://api.cow.fi/${COWSWAP_SUPPORTED_NETWORK[chainId]}/api/v1/orders`
+              await fetch(`${COW_SWAP_URL}/${orderUid}`) //api for the get request
+                .then((response) => response.json())
+                .then((data) => {
+                  if (data.status === 'presignaturePending') {
+                    preSignature = [
+                      {
+                        ...transaction,
+                        value: transaction.value,
```
Comment (jfschwarz) [inline extension/src/transactionTranslations/cowswapSetPreSignature.ts:70]: - `value` should probably be fixed to `0`
- `operation` must be set to `1` (delegatecall)

File: extension/src/transactionTranslations/cowswapSetPreSignature.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+          if (orderUid && chainId) {
+            try {
+              const COW_SWAP_URL = `https://api.cow.fi/${COWSWAP_SUPPORTED_NETWORK[chainId]}/api/v1/orders`
+              await fetch(`${COW_SWAP_URL}/${orderUid}`) //api for the get request
+                .then((response) => response.json())
+                .then((data) => {
+                  if (data.status === 'presignaturePending') {
+                    preSignature = [
+                      {
+                        ...transaction,
```
Comment (samepant) [inline extension/src/transactionTranslations/cowswapSetPreSignature.ts:69]: Does this transaction need to be modified to send to the `CowswapOrderSigner` contract?

**Comment by jfschwarz** [comment]
Pilot is now ready to support delegatecalls. They will marked as such in the drawer:

image

**Review by jfschwarz — APPROVED** [review]

**Comment by juliopavila** [comment]
Simulation test: https://dashboard.tenderly.co/public/gnosisguild/zodiac-pilot/fork-simulation/2ec4eb41-d52d-41be-bbd1-576c4bea4da0

Result of the translation
image

**Review by samepant — APPROVED** [review]

**Pull request merged**